code/game/objects/items.dm

```diff
@@ -653,6 +653,79 @@
 		min_cold_protection_temperature = ICE_PLANET_MIN_COLD_PROTECTION_TEMPERATURE
 
 
+///Play small animation and jiggle when picking up an object
+/obj/item/proc/do_pickup_animation(atom/target)
+	set waitfor = FALSE
+	if(!isturf(loc))
+		return
+	var/image/pickup_animation = image(icon = src, loc = loc, layer = layer + 0.1)
+	pickup_animation.plane = GAME_PLANE
+	pickup_animation.transform *= 0.75
+	pickup_animation.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA
+	var/turf/current_turf = get_turf(src)
+	var/direction
+	var/to_x = target.pixel_x
+	var/to_y = target.pixel_y
+
+	if(!QDELETED(current_turf) && !QDELETED(target))
+		direction = get_dir(current_turf, target)
+	if(direction & NORTH)
+		to_y += 32
+	else if(direction & SOUTH)
+		to_y -= 32
+	if(direction & EAST)
+		to_x += 32
+	else if(direction & WEST)
+		to_x -= 32
+	if(!direction)
+		to_y += 16
+	flick_overlay(pickup_animation, GLOB.clients, 6)
+	var/matrix/animation_matrix = new
+	animation_matrix.Turn(pick(-30, 30))
+	animate(pickup_animation, alpha = 175, pixel_x = to_x, pixel_y = to_y, time = 3, transform = animation_matrix, easing = CUBIC_EASING)
+	sleep(1)
```

Comment:
Sleep seems acceptable here, considering it is just an animate behind, right tivi?

Reply:
No it should be three animate calls I need to fix it on tg